lib/robots/dor_repo/assembly/content_metadata_create.rb

```diff
@@ -14,19 +16,38 @@ def perform(druid)
           obj =  item(druid)
           return LyberCore::Robot::ReturnState.new(status: :skipped, note: 'object is not an item') unless obj.item? # not an item, skip
 
+          return LyberCore::Robot::ReturnState.new(status: :skipped, note: 'No contentMetadata to load from the filesystem') if !content_metadata_exists?(obj) && !obj.stub_content_metadata_exists?
+
           # both stub and regular content metadata exist -- this is an ambiguous situation and generates an error
-          raise "#{Settings.assembly.stub_cm_file_name} and #{Settings.assembly.cm_file_name} both exist for #{druid}" if obj.stub_content_metadata_exists? && obj.content_metadata_exists?
+          raise "#{Settings.assembly.stub_cm_file_name} and #{Settings.assembly.cm_file_name} both exist for #{druid}" if obj.stub_content_metadata_exists? && content_metadata_exists?(obj)
 
-          # regular content metadata exists -- do not recreate it
-          return LyberCore::Robot::ReturnState.new(status: :skipped, note: "#{Settings.assembly.cm_file_name} exists") if obj.content_metadata_exists?
+          # Build the xml
+          xml = if obj.stub_content_metadata_exists?
+                  obj.convert_stub_content_metadata
+                else
+                  File.read(cm_file_name(obj))
+                end
 
-          # neither stubContentMetadata or contentMetadata exist.
-          raise "Unable to find #{Settings.assembly.stub_cm_file_name} or #{Settings.assembly.cm_file_name}" unless obj.stub_content_metadata_exists?
+          # Convert the XML to cocina and save it
+          structural = Dor::StructuralMetadata.update(xml, obj.cocina_model)
+          updated = obj.cocina_model.new(structural: structural)
+          obj.object_client.update(params: updated)
+
+          # Remove the contentMetadata.xml
+          FileUtils.rm(cm_file_name(obj)) if content_metadata_exists?(obj)
 
```

Comment:
if the robot step is re-run?  i was just thinking that if the stubContentMetadata was there and you re-ran the step, it would be re-coverted to contentMetadata in line 25

Reply:
The directory will be cleaned up at the end of the accessioningWF, so I don't think that would ever happen.  How would you re-run that one step?

Reply:
Actually, I'm think I was forgetting the nature of this robot, vs the `content-metadata` step in accessionWF.  This step was intended to create contentMetadata XML if needed (either from scratch or by converting the stubContentMetadata file).  The later step in accessionWF will pick up the contentMetadata XML file (if it exists) and convert to cocina, right?  So in the case, do we need to do that here too?  or I am confused?

Reply:
I am interested in the answer to this as well -- @jcoyne could I ask you to do a PR in workflow-server-rails, changing the descriptions of appropriate steps in the assemblyWF and the accessioningWF?  I'll make a ticket and assign it to you.

Reply:
See sul-dlss/workflow-server-rails#569